### PR TITLE
Refactor: Update and simplify CLI

### PR DESCRIPTION
**Pull request by cortadocodes**
### Main changes
* Re-enable CLI
* Add `run` command to CLI that uses `Runner` to find the right `app` module and use it
* Remove `octue_version` decorator, replacing with provision of the app source directory
* Add `--version` option
* Add `-h` option for command line help
* Remove option to pass in JSON strings as inputs
* Re-enable entry point
* Test version, help, and run CLI commands

### Minor refactors
* Mirror package structure in tests package
* Remove need for `#noqa` by using `__all__`
* Simplify paths in `BaseTest`
* Remove unnecessary setter method for read-only attributes

### Still to do
* Pass in output values, output manifest, monitors, children - I'm currently unsure how to do this
* Test with manifests
* Resolve problem where the two tests of the `run` command of the CLI pass by themselves, but not when run in serial with `test_fractal_configuration`. The two tests pass when run in serial with all other tests (i.e. when `test_fractal_configuration` is disabled. I think its something to do with paths being leaked from the template setting and making their way into other tests using `AppFrom`.
* Decide on whether or not to pass extra strand data to `Runner` (see `test_exception_raised_when_extra_strand_data_present`)

### CODE REVIEW

File: tests/base.py [hunk 1]
```diff
@@ -14,8 +14,8 @@ def setUp(self):
 
         # Set up paths to the test data directory and to the app templates directory
         root_dir = os.path.dirname(os.path.abspath(__file__))
-        self.data_path = str(os.path.join(root_dir, "data", ""))
-        self.templates_path = str(os.path.join(os.path.dirname(root_dir), "octue", "templates", ""))
```
Comment (thclark) [inline tests/base.py:18]: You know when you look at your own code from last year and you're like... "What was I on?"

File: tests/data/data_dir_with_no_manifests/configuration/values.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+    "n_iterations": 5
+}
```
Comment (thclark) [inline tests/data/data_dir_with_no_manifests/configuration/values.json:3]: @cortadocodes have you done `pre-commit install`? If so, then our end-of-file fixer isn't working for json files.

File: tests/test_runner.py [hunk 3]
```diff
@@ -86,18 +86,18 @@ def fcn(analysis):
...
+    #     """
+    #     with self.assertRaises(twined.exceptions.StrandNotFound) as error:
+    #         Runner(
+    #             twine="{}", configuration_values={},
+    #         )
+    #
+    #     self.assertIn(
+    #         "Source data is provided for 'configuration_values' but no such strand is defined in the twine",
+    #         error.exception.args[0],
+    #     )
```
Comment (thclark) [inline tests/test_runner.py:100]: Is this no longer compatible with the latest version of twined?

**Review by thclark — APPROVED** [review]
Thanks @cortadocodes great quality work as always. I'm going to merge this in and get 0.1.4 over the finish line :)

File: tests/data/data_dir_with_no_manifests/configuration/values.json (hunk 2 above)
Comment (cortadocodes) [reply tests/data/data_dir_with_no_manifests/configuration/values.json:3]: I think I may not have done for `octue-sdk-python` - I'll do it now

Comment (cortadocodes) [reply tests/data/data_dir_with_no_manifests/configuration/values.json:3]: It seems to be skipping all the checks when I run `pre-commit`?

File: tests/test_runner.py (hunk 3 above)
Comment (cortadocodes) [reply tests/test_runner.py:100]: I think it's not valid anymore with the changes I made (I've allowed e.g. configuration values to be passed into *the Runner* even if the twine doesn't allow it to make the CLI simpler; note: the twine still behaves the same and won't allow extra strand data to be used)

Comment (cortadocodes) [reply tests/test_runner.py:100]: Shall I remove it?

**Comment by cortadocodes** [comment]
@thclark what do you think about the "still to do" section in the description above?

File: tests/test_runner.py (hunk 3 above)
Comment (thclark) [reply tests/test_runner.py:100]: I'll do it, I've got it right here

**Comment by thclark** [comment]
> @thclark what do you think about the "still to do" section in the description above? Mainly bullet points 1 and 4

Just working through now

**Pull request merged**